### PR TITLE
ci: add verify pr workflow [KHCP-10604]

### PR DESCRIPTION
**Pull request by portikM**
# Summary

Jira: https://konghq.atlassian.net/browse/KHCP-10604

Adds workflow to verify PR doesn't reference any preview packages

#### Resources

- [Creating a package docs](https://github.com/Kong/public-ui-components/blob/main/docs/creating-a-package.md)

### CODE REVIEW

**Review by adamdehaven — DISMISSED** [review]
lgtm

File: .github/workflows/pr-packages.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      - main
+
+concurrency:
+  group: ${{ github.ref }}-${{ github.workflow }}
+  cancel-in-progress: true
+
+jobs:
+  check-packages:
+    name: No PR preview references
+    runs-on: ${{ vars.RUNS_ON }}
```
Comment (ValeryG) [inline .github/workflows/pr-packages.yaml:14]:
```suggestion
    runs-on: ubuntu-latest
```
we do not run public on hosted runners and do not have var for it

Comment (adamdehaven) [inline .github/workflows/pr-packages.yaml:14]:
```suggestion
    runs-on: ${{ vars.RUNS_ON }}
    timeout-minutes: 5
```

Comment (portikM) [reply .github/workflows/pr-packages.yaml]: Applied

Comment (portikM) [reply .github/workflows/pr-packages.yaml]: Done

**Review by adamdehaven — APPROVED** [review]
lgtm

**Pull request merged**